### ISSUE TITLE
AUR Package

### ISSUE THREAD

**Issue by proudmuslim-dev** [inclement/vivarium#3]
Just came across this project and figured I'd open an issue for an AUR package, as packaging it there would bring obvious benefits

**Comment by lmartinez-mirror** [comment]
@proudmuslim-dev Hi, I'm currently working on packaging this for the AUR. I'll be sure to post here again when it's finished.

**Comment by proudmuslim-dev** [comment]
@lmartinez-mirror Awesome, thank you

**Comment by lmartinez-mirror** [comment]
Got a compiled binary from the PKGBUILD, but it doesn't run. I wind up with this error:
```
vivarium: symbol lookup error: vivarium: undefined symbol: wlr_xdg_surface_for_each_popup
```
@inclement Do you know what might be happening here? I'm using `wlroots` 0.12.0; master branch will not compile.

**Comment by inclement** [comment]
Interesting, I'm not sure what's up here. You'll definitely need wlroots 0.12.0 as there are significant breaking changes between versions, but I don't see why that symbol should cause problems.

What is the output of `nm --format sysv -D /path/to/libwlroots.so.7 | grep wlr_xdg_surface_for_each_popup`? And indeed, what is that path? I'm developing on arch with the latest wlroots 0.12.0-1 package, which has this symbol.

**Comment by lmartinez-mirror** [comment]
Sorry, I forgot I use `wlroots-git` on my personal system, so of course it won't run!

`nm` and `grep` successfully find this symbol in `vivarium` in a clean chroot. I'll upload the PKGBUILD here for anyone to test before I submit it to the AUR.

**Comment by inclement** [comment]
Ah great, thanks, happy to test the PKGBUILD.

**Comment by lmartinez-mirror** [comment]
[PKGBUILD.zip](https://github.com/inclement/vivarium/files/6151958/PKGBUILD.zip)

**Comment by inclement** [comment]
Works great. May I suggest s/wlroots/wayland in the pkgdesc?

    < pkgdesc="A dynamic tiling wlroots compositor inspired by xmonad"
    ---
    > pkgdesc="A dynamic tiling wayland compositor inspired by xmonad"

**Comment by lmartinez-mirror** [comment]
Sure thing, I'll make that change before I push.

@proudmuslim-dev Tagging because I know you were interested in a PKGBUILD, it'll go live soon.

**Comment by lmartinez-mirror** [comment]
[AUR package is live.](https://aur.archlinux.org/packages/vivarium-git/) I'll create another PKGBUILD when Vivarium hits its first stable release. This issue can be closed.

**Comment by proudmuslim-dev** [comment]
@lmartinez-mirror Awesome, closing the issue now

**Comment by ProjectInfinity** [comment]
How would you go about this now that arch ships wlroots 0.12? The pkgbuild seem to fail for me.

**Comment by inclement** [comment]
I've added an AUR comment explaining why it's failing, and a couple of workarounds.

In short: https://github.com/inclement/vivarium/pull/43 should work with wlroots 0.13 (but isn't well tested yet), or if you manually build then it will automatically download and build against a compatible wlroots version (currently 0.12.0).

**Comment by lmartinez-mirror** [comment]
Thanks for the heads up, I was fixing to raise a new issue about this. I'll update it as soon as it's merged.

**Comment by inclement** [comment]
Sorry for the delay there, took longer than expected due to focusing on damage tracking (which works now!).

I've just merged #43 which should mean everything works fine with wlroots 0.13.0 now.
